pkg/services/ngalert/api/tooling/definitions/cortex-ruler.go

```diff
@@ -396,3 +394,70 @@ type GettableGrafanaRule struct {
 	Provenance      Provenance          `json:"provenance,omitempty" yaml:"provenance,omitempty"`
 	IsPaused        bool                `json:"is_paused" yaml:"is_paused"`
 }
+
+// AlertQuery represents a single query associated with an alert definition.
+type AlertQuery struct {
+	// RefID is the unique identifier of the query, set by the frontend call.
+	RefID string `json:"refId"`
+	// QueryType is an optional identifier for the type of query.
+	// It can be used to distinguish different types of queries.
+	QueryType string `json:"queryType"`
+	// RelativeTimeRange is the relative Start and End of the query as sent by the frontend.
+	RelativeTimeRange RelativeTimeRange `json:"relativeTimeRange"`
+
+	// Grafana data source unique identifier; it should be '__expr__' for a Server Side Expression operation.
+	DatasourceUID string `json:"datasourceUid"`
+
+	// JSON is the raw JSON query and includes the above properties as well as custom properties.
+	Model json.RawMessage `json:"model"`
```

Comment:
Just confirming - the fact that we never mapped the private `modelProps` isn't an issue right? The query definitions still contain the expected `intervalMs`/`maxDataPoints` because the real source of truth for these is still kept in the main model - right? Just wanting to make sure we aren't shedding data implicitly when we convert types.